bbq/vm/vm.go

```diff
@@ -797,9 +797,11 @@ func opNot(vm *VM) {
 
 func opUnwrap(vm *VM) {
 	value := vm.peek()
-	if someValue, ok := value.(*SomeValue); ok {
-		value = someValue.value
-		vm.replaceTop(value)
+	switch value := value.(type) {
+	case *SomeValue:
+		vm.replaceTop(value.value)
+	case NilValue:
+		panic(ForceNilError{})
```

Comment:
Maybe panic in the `default` case with an unreachable-error

Reply:
Added in [ff2aec9](https://github.com/onflow/cadence/pull/3803/commits/ff2aec9ee3658020ce360f316016d0cd1570b90e).

This made me realize that we actually allow force-unwrapping of non-optional values, so I adjusted compilation to support it (omit unwrap if not needed) in [a55dfe5](https://github.com/onflow/cadence/pull/3803/commits/a55dfe5e0563d8d68684000987cd7416f9f65544)